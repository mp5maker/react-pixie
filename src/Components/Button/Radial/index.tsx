import * as React from 'react'
import PropTypes from 'prop-types'
import { motion } from 'framer-motion'

import './styles.scss'

interface ButtonRadialPropsInterface {
    style?: any,
    onClick?: ((params: any) => any) | ((params: any) => void),
    children: React.ReactNode,
    variants?: any,
    initial?: string,
    animate?: string,
    exit?: string,
}

export const ButtonRadial: React.FC<ButtonRadialPropsInterface> = ({ style = {}, onClick = () => {}, children, ...otherProps }: any = {}) => {
    return (
        <motion.button
            className={`button-radial-container`}
            style={style}
            onClick={onClick}
            {...otherProps}>
            { children }
        </motion.button>
    )
}


ButtonRadial.propTypes = {
    style: PropTypes.shape({}),
    onClick: PropTypes.func,
    children: PropTypes.node,
    variants: PropTypes.shape({}),
    initial: PropTypes.string,
    animate: PropTypes.string,
    exit: PropTypes.string,
}