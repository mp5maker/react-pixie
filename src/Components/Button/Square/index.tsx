import * as React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import './styles.scss';

interface ButtonSquarePropsInterface {
  style?: any;
  onClick?: ((params: any) => any) | ((params: any) => void);
  children: React.ReactNode;
}

export const ButtonSquare: React.FC<ButtonSquarePropsInterface> = ({
  style = {},
  onClick = () => {},
  children,
}: any = {}) => {
  return (
    <Button
      className={`button-square-container`}
      style={style}
      onClick={onClick}>
      {children}
    </Button>
  );
};

ButtonSquare.propTypes = {
  style: PropTypes.shape({}),
  onClick: PropTypes.func,
  children: PropTypes.node,
};
