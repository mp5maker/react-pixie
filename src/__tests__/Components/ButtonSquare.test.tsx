import React from 'react'
import { shallow, render, mount } from 'enzyme'
import sinon from 'sinon'

import { ButtonSquare } from '../../Components/Button/Square'

describe(`Button Square`, () => {
    it('It should shallow properly', () => {
        const wrapper = shallow(
            <ButtonSquare
                style={{}}
                onClick={() => {}}>
                Test Button
            </ButtonSquare>
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('It should trigger clicks properly', () => {
        const spy = sinon.spy()

        const wrapper = mount(
            <ButtonSquare
                style={{}}
                onClick={spy}>
                Test Button
            </ButtonSquare>
        )

        wrapper
            .find('.button-square-container')
            .first()
            .simulate('click')

        expect(spy.calledOnce).toBe(true)
    })
})