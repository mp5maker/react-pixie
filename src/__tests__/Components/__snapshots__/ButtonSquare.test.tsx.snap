// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Square It should shallow properly 1`] = `
<WithStyles(ForwardRef(Button))
  className="button-square-container"
  onClick={[Function]}
  style={Object {}}
>
  Test Button
</WithStyles(ForwardRef(Button))>
`;
